mem/module_info.c

```diff
@@ -25,7 +25,11 @@
 #ifdef SHM_EXTRA_STATS
 
 #include <dlfcn.h>
+#ifdef HAVE_STDATOMIC
 #include <stdatomic.h>
+#else
+#include "../atomic.h"
+#endif
```

Comment:
On a first thought, maybe we could put this cookie-cutter "#ifdef" code into some common header and just include that one.

However, the more I think about it, the more I like your way better, because it **screams** that it needs simplification to a single line in some distant future, when we will just drop support for gcc 4.8 and below.